src/middleware/api/methods.js

```diff
@@ -0,0 +1,64 @@
+const last = require('lodash/last');
+const { conj, effect } = require('src/utils');
+const { omitReadOnly, defaults, validate } = require('src/schema-utils');
+
+
+function create(schema, fn) {
+  return method((ctx, args, next) => Promise.resolve(ctx.request.body)
+    .then(effect(d => validate(schema, d)))
+    .then(d => defaults(schema, d))
+    .then(d => fn(...args, d))
+    .then(res => { ctx.body = res; })
+    .then(() => next()));
+}
+
+
+function read(fn, defaults = () => ({})) {
+  return method((ctx, args, next) => Promise.resolve(ctx.request.query)
+    .then(d => conj(defaults(), d))
+    .then(d => fn(...args, d))
+    .then(res => { ctx.body = res; })
+    .then(() => next()));
+}
+
+
+function update(schema, fn) {
+  return method((ctx, args, next) => Promise.resolve(ctx.request.body)
+    .then(d => omitReadOnly(schema, d))
+    .then(effect(d => validate(schema, d)))
+    .then(d => defaults(schema, d))
+    .then(d => fn(...args, d))
+    .then(res => { ctx.body = res; })
```

Comment:
Thanks for catching this! I think only `create` methods will need to set the response code to `201`, will update `create()` to do this.